userspace/libscap/scap_procs.c

```diff
@@ -998,7 +1012,7 @@ static int32_t _scap_proc_scan_proc_dir_impl(scap_t* handle, char* procdirname,
 		//
 		// We have a process that needs to be explored
 		//
-		res = scap_proc_add_from_proc(handle, tid, procdirname, &sockets_by_ns, NULL, add_error);
+		res = scap_proc_add_from_proc(handle, tid, procdirname, &sockets_by_ns, NULL, suppress, add_error);
 		if(res != SCAP_SUCCESS)
 		{
 			snprintf(error, SCAP_LASTERR_SIZE, "cannot add procs tid = %"PRIu64", parenttid = %"PRIi32", dirname = %s, error=%s", tid, parenttid, procdirname, add_error);
```

Comment:
Rather than suppressing the error return from this function, I suggest we use the `suppress` parameter to decide whether to break from the `while` loop or continue. That change is less intrusive (because we don't need to change `scap_proc_add_from_proc`).